client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClient.java

```diff
@@ -215,20 +285,99 @@ private WinRm getService() {
-            bp.getRequestContext().put(BindingProvider.PASSWORD_PROPERTY, password);
+
+        switch(authenticationScheme) {
+            case AuthSchemes.BASIC:
+                bp.getRequestContext().put(BindingProvider.USERNAME_PROPERTY, username);
+                bp.getRequestContext().put(BindingProvider.PASSWORD_PROPERTY, password);
+            case AuthSchemes.NTLM:
+                Credentials creds = new NTCredentials(username, password, null, null);
+
+                CredentialsProvider credsProvider = new BasicCredentialsProvider();
+                credsProvider.setCredentials(
+                        new AuthScope(AuthScope.ANY_HOST, AuthScope.ANY_PORT),
+                        creds);
+
+                Registry<AuthSchemeProvider> authSchemeRegistry = RegistryBuilder.<AuthSchemeProvider>create()
+                        .register(AuthSchemes.BASIC, new BasicSchemeFactory())
+                        // Uncomment to support Negotiate(SPNEGO)+Kerberos, only one of the next two items allowed
+                        // .register(AuthSchemes.SPNEGO, new SPNegoSchemeFactory())
+                        .register(AuthSchemes.SPNEGO, new SpNegoNTLMSchemeFactory())
+                        .register(AuthSchemes.KERBEROS, new KerberosSchemeFactory())//
+                        .build();
+
+                bp.getRequestContext().put(Credentials.class.getName(), creds);
+                bp.getRequestContext().put("http.autoredirect", true);
+
+                bp.getRequestContext().put(AuthSchemeProvider.class.getName(), authSchemeRegistry);
+                HTTPClientPolicy httpClientPolicy = new HTTPClientPolicy();
+//            httpClientPolicy.setConnectionTimeout(36000);
+                httpClientPolicy.setAllowChunking(false);
+
+                AsyncHTTPConduit httpClient = (AsyncHTTPConduit) client.getConduit();
+
+                TLSClientParameters tlsClientParameters = new TLSClientParameters();
+                if (disableCertificatesChecks) {
+                    tlsClientParameters.setDisableCNCheck(true);
+                    tlsClientParameters.setTrustManagers(new TrustManager[]{new X509TrustManager() {
+                        @Override
+                        public void checkClientTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException {
+
+                        }
+
+                        @Override
+                        public void checkServerTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException {
+
+                        }
+
+                        @Override
+                        public X509Certificate[] getAcceptedIssuers() {
+                            return new X509Certificate[0];
+                        }
+                    }});
+                }
+                httpClient.setTlsClientParameters(tlsClientParameters);
+
+                httpClient.setClient(httpClientPolicy);
+                httpClient.getClient().setAutoRedirect(true);
```

Comment:
Probably just one of 'http.autoredirect` and `setAutoRedirect` needed? Though I haven't seen redirects in winrm so it's just in case?